lightning/src/routing/router.rs

```diff
@@ -333,84 +592,262 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, targ
 		};
 	}
 
-	match network.get_nodes().get(target) {
-		None => {},
-		Some(node) => {
-			add_entries_to_cheapest_to_target_node!(node, target, 0);
-		},
-	}
+	let mut payment_paths = Vec::<PaymentPath>::new();
+
+	// TODO: diversify by nodes (so that all paths aren't doomed if one node is offline).
+	'paths_collection: loop {
+		// For every new path, start from scratch, except
+		// bookkeeped_channels_liquidity_available_msat, which will improve
+		// the further iterations of path finding. Also don't erase first_hop_targets.
+		targets.clear();
+		dist.clear();
```

Comment:
Should we fail routing if one of those maps grow dangerously towards a memory DoS ?

In the future, it's likely mobile clients will get their network graph from some semi-trusted sources. This network graph will be assumed to be valid. This graph might have unbounded dimensions.

Can an attacker a) feed a client with such a malicious graph which will b) trigger a payment toward some payee such as c) crash the payer ?

Reply:
Not sure this is the right place to handle that problem.

Reply:
Yeah I had the discussion with Matt on IRC (http://gnusha.org/rust-bitcoin/2021-02-11.log, at 15:19). It would be handled in devs documentation. Up to them to sanitize the graph if they get it from untrusted sources.

Mentioned in #565